### PR TITLE
feat(auth): Add bulk get/delete methods

### PR DESCRIPTION
**Pull request by rsgowman**
This PR allows callers to retrieve a list of users by unique identifier (uid, email, phone, federated provider uid) as well as to delete a list of users.

RELEASE NOTE: Added getUsers() and deleteUsers() APIs for retrieving and deleting user accounts in bulk.

### CODE REVIEW

File: src/main/java/com/google/firebase/auth/FirebaseUserManager.java [hunk 1]
```diff
@@ -171,6 +178,37 @@ UserRecord getUserByPhoneNumber(String phoneNumber) throws FirebaseAuthException
     return new UserRecord(response.getUsers().get(0), jsonFactory);
   }
 
+  /**
+   * @pre identifiers != null
```
Comment (rsgowman) [inline src/main/java/com/google/firebase/auth/FirebaseUserManager.java]: These are verified in the calling code. (As an alternative, we could just repeat the validation here.)

File: src/test/java/com/google/firebase/auth/FirebaseAuthIT.java [hunk 2]
```diff
@@ -137,6 +137,83 @@ public void testDeleteNonExistingUser() throws Exception {
...
+    assertEquals(0, result.getFailureCount());
+    assertTrue(result.getErrors().isEmpty());
+  }
+
+  /**
+   * The batchDelete endpoint is currently rate limited to 1qps. Use this test helper to ensure we
+   * don't run into quota exceeded errors.
+   */
+  // TODO(rsgowman): When/if the rate limit is relaxed, eliminate this helper.
+  private ApiFuture<DeleteUsersResult> slowDeleteUsersAsync(List<String> uids) throws Exception {
```
Comment (rsgowman) [inline src/test/java/com/google/firebase/auth/FirebaseAuthIT.java]: We may need to do something similar for the other ports... There seems to be some elasticity such that the full test suite can often pass, but not if you run just the delete users tests in quick succession (as I did while testing).

File: src/test/java/com/google/firebase/auth/GetUsersIT.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
+/*
```
Comment (rsgowman) [inline src/test/java/com/google/firebase/auth/GetUsersIT.java]: junit4 doesn't allow nested tests (and therefore doesn't allow nested setup/teardown functions.) The recommendation is to just split into a separate file, which is what I've done here.

File: src/main/java/com/google/firebase/auth/DeleteUsersResult.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+import com.google.firebase.auth.internal.BatchDeleteResponse;
+import com.google.firebase.internal.NonNull;
+import java.util.List;
+
+/**
+ * Represents the result of the {@link FirebaseAuth#deleteUsersAsync(List)} API.
+ */
+public final class DeleteUsersResult {
+
+  private int successCount;
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/DeleteUsersResult.java]: Make all these `final` as well.

File: src/main/java/com/google/firebase/auth/DeleteUsersResult.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+   */ 
+  public int getSuccessCount() {
+    return successCount;
+  }
+
+  /**
+   * Returns the number of users that failed to be deleted (possibly zero).
+   */
+  public int getFailureCount() {
+    return failureCount;
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/DeleteUsersResult.java]: Perhaps remove this attribute and just `return errors.size()`

File: src/main/java/com/google/firebase/auth/EmailIdentifier.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Used for looking up an account by email.
+ *
+ * @see {FirebaseAuth#getUsers}
+ */
+public class EmailIdentifier extends UserIdentifier {
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/EmailIdentifier.java]: final

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java [hunk 7]
```diff
@@ -600,6 +603,81 @@ protected UserRecord execute() throws FirebaseAuthException {
     };
   }
 
+  /**
+   * Gets the user data corresponding to the specified identifiers.
+   *
+   * <p>There are no ordering guarantees; in particular, the nth entry in the users result list is
+   * not guaranteed to correspond to the nth entry in the input parameters list.
+   *
+   * <p>Only a maximum of 100 identifiers may be supplied. If more than 100 identifiers are
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/FirebaseAuth.java]: Nit: s/supplied/specified

File: src/main/java/com/google/firebase/auth/FirebaseUserManager.java (hunk 1 above)
Comment (hiranya911) [reply src/main/java/com/google/firebase/auth/FirebaseUserManager.java]: Let's drop the `pre` annotations. That's not really a convention we use. It is understood that the arguments passed to this class have already been validated. If you want add a class-level javadoc to indicate that explicitly.

File: src/main/java/com/google/firebase/auth/GetUsersResult.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.firebase.internal.NonNull;
+import java.util.Set;
+
+/**
+ * Represents the result of the {@link FirebaseAuth#getUsersAsync(Collection)} API.
+ */
+public final class GetUsersResult {
+  private Set<UserRecord> users;
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/GetUsersResult.java]: final

File: src/main/java/com/google/firebase/auth/GetUsersResult.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  @NonNull
+  public Set<UserRecord> getUsers() {
+    return this.users;
+  }
+
+  /**
+   * Set of identifiers that were requested, but not found.
+   */
+  @NonNull
+  Set<UserIdentifier> getNotFound() {
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/GetUsersResult.java]: Should be public?

File: src/main/java/com/google/firebase/auth/UidIdentifier.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+import com.google.firebase.auth.internal.GetAccountInfoRequest;
+import com.google.firebase.internal.NonNull;
+
+/**
+ * Used for looking up an account by uid.
+ *
+ * @see {FirebaseAuth#getUsers}
+ */
+public class UidIdentifier extends UserIdentifier {
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/UidIdentifier.java]: final -- here and other similar classes.

File: src/main/java/com/google/firebase/auth/UserIdentifier.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+package com.google.firebase.auth;
+
+import com.google.firebase.auth.internal.GetAccountInfoRequest;
+import com.google.firebase.internal.NonNull;
+
+/**
+ * Identifies a user to be looked up.
+ */
+public abstract class UserIdentifier {
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/UserIdentifier.java]: Might be too late for this conversation. But I wonder if we should make all implementations of this class package-protected and provide a set of static factory methods instead.

```
public static UserIdentifier fromUid(String uid);
public static UserIdentifier fromEmail(String email);
// and so on
```

Does that make sense?

File: src/main/java/com/google/firebase/auth/UserRecord.java [hunk 12]
```diff
@@ -80,7 +81,15 @@
       }
     }
     this.tokensValidAfterTimestamp = response.getValidSince() * 1000;
-    this.userMetadata = new UserMetadata(response.getCreatedAt(), response.getLastLoginAt());
+
+    String lastRefreshAtRfc3339 = response.getLastRefreshAt();
+    long lastRefreshAtMillis = 0;
+    if (lastRefreshAtRfc3339 != null && !lastRefreshAtRfc3339.isEmpty()) {
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/UserRecord.java]: `!Strings.isNullOrEmpty()`

File: src/main/java/com/google/firebase/auth/internal/GetAccountInfoRequest.java [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+package com.google.firebase.auth.internal;
+
+import com.google.api.client.util.Key;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents the request to lookup account information.
+ */
+public class GetAccountInfoRequest {
```
Comment (hiranya911) [inline src/main/java/com/google/firebase/auth/internal/GetAccountInfoRequest.java]: Make these final too.

File: src/test/java/com/google/firebase/auth/FirebaseAuthIT.java [hunk 14]
```diff
@@ -137,6 +137,83 @@ public void testDeleteNonExistingUser() throws Exception {
...
 
+  @Test
+  public void testDeleteUsers() throws Exception {
+    UserRecord user1 = newUserWithParams();
+    UserRecord user2 = newUserWithParams();
+    UserRecord user3 = newUserWithParams();
+
+    DeleteUsersResult deleteUsersResult = slowDeleteUsersAsync(
+        ImmutableList.of(user1.getUid(), user2.getUid(), user3.getUid())
+        ).get();
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/auth/FirebaseAuthIT.java]: Nit: Kind of strange line wrapping here.

File: src/test/java/com/google/firebase/auth/FirebaseAuthIT.java [hunk 15]
```diff
@@ -248,6 +325,35 @@ public void testUserLifecycle() throws Exception {
...
+    try {
+      // New users should not have a lastRefreshTimestamp set.
+      assertEquals(0, newUserRecord.getUserMetadata().getLastRefreshTimestamp());
+
+      // Login to cause the lastRefreshTimestamp to be set.
+      signInWithPassword(newUserRecord.getEmail(), "password");
+
+      UserRecord userRecord = auth.getUser(newUserRecord.getUid());
+
+      // Ensure the lastRefreshTimestamp is approximately "now" (with a tollerance of 30 minutes).
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/auth/FirebaseAuthIT.java]: 30 minutes sound too high. Did you mean 30 seconds?

File: src/test/java/com/google/firebase/auth/GetUsersIT.java [hunk 16]
```diff
@@ -0,0 +1,152 @@
...
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class GetUsersIT {
+  private static FirebaseAuth auth;
+  private static UserRecord testUser1;
+  private static UserRecord testUser2;
+  private static UserRecord testUser3;
+  private static String importUser1Uid;
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/auth/GetUsersIT.java]: importUserUid

File: src/main/java/com/google/firebase/auth/DeleteUsersResult.java [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+      }
+    }
+    errors = errorsBuilder.build();
+    failureCount = errors.size();
+    successCount = users - errors.size();
+  }
+
+  /**
+   * Returns the number of users that were deleted successfully (possibly zero). Users that did
+   * not exist prior to calling deleteUsersAsync() will be considered to be successfully
```
Comment (egilmorez) [inline src/main/java/com/google/firebase/auth/DeleteUsersResult.java]: Suggest code font/link for the method and present tense:

{@link deleteUsersAsync} are considered

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java (hunk 7 above)
Comment (egilmorez) [inline src/main/java/com/google/firebase/auth/FirebaseAuth.java]: Suggest just "A maximum . . . "

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java [hunk 18]
```diff
@@ -600,6 +603,81 @@ protected UserRecord execute() throws FirebaseAuthException {
...
+   * supplied, this method will immediately throw an IllegalArgumentException.
```
Comment (egilmorez) [inline src/main/java/com/google/firebase/auth/FirebaseAuth.java]: Suggest "this method throws an {@link ...}

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java [hunk 19]
```diff
@@ -600,6 +603,81 @@ protected UserRecord execute() throws FirebaseAuthException {
...
+    return getUsersOp(identifiers).call();
+  }
+
+  /**
+   * Gets the user data corresponding to the specified identifiers.
+   *
+   * <p>There are no ordering guarantees; in particular, the nth entry in the users result list is
+   * not guaranteed to correspond to the nth entry in the input parameters list.
+   *
+   * <p>Only a maximum of 100 identifiers may be supplied. If more than 100 identifiers are
```
Comment (egilmorez) [inline src/main/java/com/google/firebase/auth/FirebaseAuth.java]: If we say "specified" above, this should probably work that way too.

Same on the present tense/link format in line 635.

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java [hunk 20]
```diff
@@ -841,6 +919,65 @@ protected Void execute() throws FirebaseAuthException {
...
   }
 
+  /**
+   * Deletes the users specified by the given identifiers.
+   *
+   * <p>Deleting a non-existing user won't generate an error. (i.e. this method is idempotent.)
+   * Non-existing users will be considered to be successfully deleted, and will therefore be counted
+   * in the DeleteUsersResult.getSuccessCount() value.
+   *
+   * <p>Only a maximum of 1000 identifiers may be supplied. If more than 1000 identifiers are
```
Comment (egilmorez) [inline src/main/java/com/google/firebase/auth/FirebaseAuth.java]: Similar changes here and next line as in 634 and 635.

File: src/main/java/com/google/firebase/auth/GetUsersResult.java [hunk 21]
```diff
@@ -0,0 +1,52 @@
...
+  private Set<UserRecord> users;
+  private Set<UserIdentifier> notFound;
+
+  GetUsersResult(@NonNull Set<UserRecord> users, @NonNull Set<UserIdentifier> notFound) {
+    this.users = checkNotNull(users);
+    this.notFound = checkNotNull(notFound);
+  }
+
+  /**
+   * Set of user records, corresponding to the set of users that were requested. Only users
```
Comment (egilmorez) [inline src/main/java/com/google/firebase/auth/GetUsersResult.java]: Suggest omitting this comma.

File: src/main/java/com/google/firebase/auth/internal/BatchDeleteResponse.java [hunk 22]
```diff
@@ -0,0 +1,51 @@
...
+ * limitations under the License.
+ */
+
+package com.google.firebase.auth.internal;
+
+import com.google.api.client.util.Key;
+import java.util.List;
+
+/**
+ * Represents the response from identity toolkit for a batch delete request.
```
Comment (egilmorez) [inline src/main/java/com/google/firebase/auth/internal/BatchDeleteResponse.java]: Should this be "the identity toolkit?"

The article would normally be omitted only if this is a name, like uppercase "Identity Toolkit"

File: src/main/java/com/google/firebase/auth/internal/BatchDeleteResponse.java [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  public static class ErrorInfo {
+    @Key("index")
+    private int index;
+
+    @Key("message")
+    private String message;
+
+    // A 'localId' field also exists here, but is not currently exposed in the admin sdk.
```
Comment (egilmorez) [inline src/main/java/com/google/firebase/auth/internal/BatchDeleteResponse.java]: Suggest caps for "Admin SDK."

File: src/test/java/com/google/firebase/auth/FirebaseAuthIT.java [hunk 24]
```diff
@@ -137,6 +137,83 @@ public void testDeleteNonExistingUser() throws Exception {
...
+        ImmutableList.of(user1.getUid())
+        ).get();
+
+    assertEquals(1, result.getSuccessCount());
+    assertEquals(0, result.getFailureCount());
+    assertTrue(result.getErrors().isEmpty());
+  }
+
+  /**
+   * The batchDelete endpoint is currently rate limited to 1qps. Use this test helper to ensure we
```
Comment (egilmorez) [inline src/test/java/com/google/firebase/auth/FirebaseAuthIT.java]: Suggest {@code batchDelete} 

(or @link if that actually works for this)

File: src/test/java/com/google/firebase/auth/FirebaseAuthIT.java [hunk 25]
```diff
@@ -248,6 +325,35 @@ public void testUserLifecycle() throws Exception {
...
+  public void testLastRefreshTime() throws Exception {
+    RandomUser user = RandomUser.create();
+    UserRecord newUserRecord = auth.createUser(new CreateRequest()
+        .setUid(user.uid)
+        .setEmail(user.email)
+        .setEmailVerified(false)
+        .setPassword("password"));
+
+    try {
+      // New users should not have a lastRefreshTimestamp set.
```
Comment (egilmorez) [inline src/test/java/com/google/firebase/auth/FirebaseAuthIT.java]: Suggest @link or @code for all instances of this.

File: src/main/java/com/google/firebase/auth/UserIdentifier.java (hunk 11 above)
Comment (rsgowman) [reply src/main/java/com/google/firebase/auth/UserIdentifier.java]: Discussed offline. tl;dr: for better or worse, we're going to keep it as is for now.

File: src/test/java/com/google/firebase/auth/FirebaseAuthIT.java (hunk 15 above)
Comment (rsgowman) [reply src/test/java/com/google/firebase/auth/FirebaseAuthIT.java]: I did mean 30m. I considered something shorter (such as 30s) but I wanted to allow a bit of clock skew as well as time for an extra long context switch on an overloaded CI server. (Both of which have bitten me in the past.) 30m seems like it should be more than long enough to cover both goals, but short enough that 
a) a random value is highly unlikely to fall within the range.
b) timezone incorrectness should still cause a failure. Hm. Well, possibly except for "interesting" timezones (i.e. non-hour increments from UTC), I suppose. I've reduced the timeout from 30m to 10m, which should put us beneath a timezone error from a +Xh:45m timezone.

File: src/test/java/com/google/firebase/auth/GetUsersIT.java (hunk 16 above)
Comment (rsgowman) [reply src/test/java/com/google/firebase/auth/GetUsersIT.java]: Done.

File: src/test/java/com/google/firebase/auth/FirebaseAuthIT.java (hunk 14 above)
Comment (rsgowman) [reply src/test/java/com/google/firebase/auth/FirebaseAuthIT.java]: hmm... we should hook up clang-format or similar. I've ran clang-format over this file (just my changes; separately from the other review feedback).

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java [hunk 26]
```diff
@@ -600,6 +603,81 @@ protected UserRecord execute() throws FirebaseAuthException {
...
+    for (UserRecord userRecord : userRecords) {
+      if (id.matches(userRecord)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
   /**
    * Gets a page of users starting from the specified {@code pageToken}. Page size will be
```
Comment (RelaxedPear) [inline src/main/java/com/google/firebase/auth/FirebaseAuth.java]: "Page size is limited to..."

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java [hunk 27]
```diff
@@ -841,6 +919,65 @@ protected Void execute() throws FirebaseAuthException {
     };
   }
 
+  /**
+   * Deletes the users specified by the given identifiers.
+   *
+   * <p>Deleting a non-existing user won't generate an error. (i.e. this method is idempotent.)
```
Comment (RelaxedPear) [inline src/main/java/com/google/firebase/auth/FirebaseAuth.java]: Suggest:

"Deleting a non-existing user does not generate an error (the method is idempotent). Non-existing users are considered to be successfully deleted and are therefore included in the `DeleteUsersResult.getSuccessCount()` value."

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java [hunk 28]
```diff
@@ -841,6 +919,65 @@ protected Void execute() throws FirebaseAuthException {
...
+   * Non-existing users will be considered to be successfully deleted, and will therefore be counted
+   * in the DeleteUsersResult.getSuccessCount() value.
+   *
+   * <p>A maximum of 1000 identifiers may be supplied. If more than 1000 identifiers are
+   * supplied, this method throws an {@link IllegalArgumentException}.
+   *
+   * <p>This API is currently rate limited at the server to 1 QPS. If you exceed this, you may get a
```
Comment (RelaxedPear) [inline src/main/java/com/google/firebase/auth/FirebaseAuth.java]: Suggest: 

"This API has a rate limit of 1 QPS. Exceeding the limit may result in a quota exceeded error. If you want to delete more than 1000 users, we suggest adding a delay to ensure you don't exceed the limit."

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java [hunk 29]
```diff
@@ -841,6 +919,65 @@ protected Void execute() throws FirebaseAuthException {
...
+      @Override
+      protected DeleteUsersResult execute() throws FirebaseAuthException {
+        return userManager.deleteUsers(uids);
+      }
+    };
+  }
+
   /**
    * Imports the provided list of users into Firebase Auth. At most 1000 users can be imported at a
    * time. This operation is optimized for bulk imports and will ignore checks on identifier
```
Comment (RelaxedPear) [inline src/main/java/com/google/firebase/auth/FirebaseAuth.java]: Suggest: 

"Imports the provided list of users into Firebase Auth. You can import a maximum of 1000 users at a time. This operation is optimized for bulk imports and ignores checks on user identifiers."

(Not completely sure about the user identifiers part, I'm assuming that's what the operation would've checked for)

File: src/main/java/com/google/firebase/auth/internal/GetAccountInfoRequest.java [hunk 30]
```diff
@@ -0,0 +1,80 @@
...
+ */
+
+package com.google.firebase.auth.internal;
+
+import com.google.api.client.util.Key;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents the request to lookup account information.
```
Comment (RelaxedPear) [inline src/main/java/com/google/firebase/auth/internal/GetAccountInfoRequest.java]: "look up" (verb)

File: src/test/java/com/google/firebase/auth/FirebaseAuthIT.java [hunk 31]
```diff
@@ -137,6 +137,78 @@ public void testDeleteNonExistingUser() throws Exception {
...
+  public void testDeleteUsersIsIdempotent() throws Exception {
+    UserRecord user1 = newUserWithParams();
+
+    DeleteUsersResult result = slowDeleteUsersAsync(ImmutableList.of(user1.getUid())).get();
+
+    assertEquals(1, result.getSuccessCount());
+    assertEquals(0, result.getFailureCount());
+    assertTrue(result.getErrors().isEmpty());
+
+    // Delete the user again, ensuring that everything still counts as a success.
```
Comment (RelaxedPear) [inline src/test/java/com/google/firebase/auth/FirebaseAuthIT.java]: Suggest: "Delete the user again to ensure that..."

File: src/test/java/com/google/firebase/auth/FirebaseAuthIT.java [hunk 32]
```diff
@@ -137,6 +137,78 @@ public void testDeleteNonExistingUser() throws Exception {
...
+    result = slowDeleteUsersAsync(ImmutableList.of(user1.getUid())).get();
+
+    assertEquals(1, result.getSuccessCount());
+    assertEquals(0, result.getFailureCount());
+    assertTrue(result.getErrors().isEmpty());
+  }
+
+  /**
+   * The {@code batchDelete} endpoint is currently rate limited to 1qps. Use this test helper to
```
Comment (RelaxedPear) [inline src/test/java/com/google/firebase/auth/FirebaseAuthIT.java]: "The {@code batchDelete} endpoint has a rate limit of 1 QPS. Use this test helper to ensure you don't exceed the quota."

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java (hunk 26 above)
Comment (rsgowman) [reply src/main/java/com/google/firebase/auth/FirebaseAuth.java:682]: Done.

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java (hunk 27 above)
Comment (rsgowman) [reply src/main/java/com/google/firebase/auth/FirebaseAuth.java:925]: Done.

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java (hunk 28 above)
Comment (rsgowman) [reply src/main/java/com/google/firebase/auth/FirebaseAuth.java:932]: Done.

File: src/main/java/com/google/firebase/auth/FirebaseAuth.java (hunk 29 above)
Comment (rsgowman) [reply src/main/java/com/google/firebase/auth/FirebaseAuth.java:983]: I changed the first two sentences as per your suggestion. I've changed the last to "This operation is optimized for bulk imports and does not check identifier uniqueness which could result in duplications".

File: src/main/java/com/google/firebase/auth/internal/GetAccountInfoRequest.java (hunk 30 above)
Comment (rsgowman) [reply src/main/java/com/google/firebase/auth/internal/GetAccountInfoRequest.java:24]: Done.

File: src/test/java/com/google/firebase/auth/FirebaseAuthIT.java (hunk 31 above)
Comment (rsgowman) [reply src/test/java/com/google/firebase/auth/FirebaseAuthIT.java:194]: Done.

File: src/test/java/com/google/firebase/auth/FirebaseAuthIT.java (hunk 32 above)
Comment (rsgowman) [reply src/test/java/com/google/firebase/auth/FirebaseAuthIT.java:203]: Done.

**Pull request merged**